### PR TITLE
Adds new chem found in roses, rose oil

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
## About the PR 
This pr adds a new chemical to roses, rose oil. This chemical when applied will increase your hygiene, if your hygiene reaches 100 it will then begin to apply the perfumed status effect. The perfumed status effect will remove miasma that is inside of you and will cancel out things that cause you to lower your hygiene meter (such as passive hygiene decay and rolling in blood), however doing so will reduce the time you have for your perfumed status. Rose Oil will also react with 2 miasma to form nothing.

## Why's this needed? 
This adds a way to counter the miasma found in rafflesia and adds a new reason to plant roses (to combat stinky crew). This should hopefully open up a lot of fun botany and crew interactions.

## Changelog

```changelog
(u)Ikea
(*)Rose oil can now be found in roses, this chemical gives you the perfumed status effect which helps counter miasma and other things that lower your hygiene meter. 1 rose oil will cancel out with 2 miasma, leaving you with an end product of nothing.
```

### CODE REVIEW

File: code/modules/status_system/statusEffects.dm [hunk 1]
```diff
@@ -1657,6 +1657,29 @@
...
+	icon_state = "perfumed"
+	unique = 1
+	maxDuration = 10 MINUTES
+
+	onUpdate()
+		if(owner.reagents.get_reagent_amount("miasma"))
+			owner.reagents.remove_reagent("miasma", 5)
+			owner.changeStatus("perfumed", -5 SECONDS)
+
+		var/mult = 1
```
Comment (TobleroneSwordfish) [inline code/modules/status_system/statusEffects.dm:1673]: not much point using `probmult` if you're just setting `mult` to 1, if you want it to be mult adjusted then use the `timePassed` parameter to `onUpdate`

**Review by TobleroneSwordfish — APPROVED** [review]
Really nice idea, better than having to stick charcoal in every positive rafflesia mix, and roses are only 2 genome points off too.

File: code/modules/status_system/statusEffects.dm (hunk 1 above)
Comment (ThatFiveGuys) [reply code/modules/status_system/statusEffects.dm]: Swapped to using prob

**Comment by DisturbHerb** [comment]
the status effect has a very big maximum duration, don't you think? ♨

**Comment by ThatFiveGuys** [comment]
This is intended to be a passive hygiene counter, not really something you actively do often. Considering that both miasma and things that lower your hygiene (such as blood) will lower the status effect a lot (and your passive decay will lower it too), I think ten minutes is a good amount. Its intentionally stronger then other hygiene reduction options because getting this requires cooperation and communication while going to a shower doesnt.

**Review by Tarmunora — CHANGES_REQUESTED** [review]
The chemical itself aside, perfume should not be able to be a wholesale replacement for personal hygiene. Act as buffer for it, sure, but you can't just febreeze yourself clean.

**Comment by ThatFiveGuys** [comment]
> The chemical itself aside, perfume should not be able to be a wholesale replacement for personal hygiene. Act as buffer for it, sure, but you can't just febreeze yourself clean.

I just think the current methods of cleanjng hygiene (press sink 50 times or stand nude in a shower) are pretty bad and this is intended to be a better alternative to that that involves actually cooperating with something

**Comment by t-toasted** [comment]
> > The chemical itself aside, perfume should not be able to be a wholesale replacement for personal hygiene. Act as buffer for it, sure, but you can't just febreeze yourself clean.
> 
> I just think the current methods of cleanjng hygiene (press sink 50 times or stand nude in a shower) are pretty bad and this is intended to be a better alternative to that that involves actually cooperating with something

I agree with Ikea. The amount of times botany will be bothered to make roses (after the newness of this PR wears off) to give people rose oil will be extremely low unless you specifically request it.

 I’d much prefer having to ask botany for roses, come back after 15 minutes and have oil, vs just spamming the medbay sink with my gloves off 50 times. 

And that’s if I even bother with the hygiene motive at all.

**Comment by UrsulaMejor** [comment]
Would prefer if this counteracted smelling bad but did nothing to hygiene. 

Hygiene mechanics aren't the best and encourage degenerate behavior like spending 5m washing your hands instead of 20s taking a shower. That said, it would be better to improve those rather than just totally making it inert with one chemical

**Comment by M-Earthfire** [comment]
I mean a good approach would be give a lower cap on hygiene of like... 20? That way, rose perfume still counteract the stinky, but if you loose it you have a small time window before you return to stinky.

**Comment by ThatFiveGuys** [comment]
> Would prefer if this counteracted smelling bad but did nothing to hygiene.
> 
> Hygiene mechanics aren't the best and encourage degenerate behavior like spending 5m washing your hands instead of 20s taking a shower. That said, it would be better to improve those rather than just totally making it inert with one chemical

Did this

File: code/datums/sims.dm [hunk 2]
```diff
@@ -246,6 +246,9 @@
 
 		var/protection = 20
 
+		modifyValue(var/amt)
+			..()
```
Comment (ZeWaka) [inline code/datums/sims.dm:250]: useless

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 3]
```diff
@@ -4279,3 +4279,10 @@ datum
 			mix_phrase = "The mixture comes together slowly. It doesn't seem like it wants to be here."
 			required_reagents = list("poor_cement" = 1, "silicon_dioxide" = 5, "water" = 1)
 			result_amount = 7
+
+		roseoil
+			name = "roseoil"
+			id = "roseoil"
+			mix_phrase = null
+			mix_sound = null
+			required_reagents = list("roseoil" = 1, "miasma" = 2)
```
Comment (ZeWaka) [inline code/modules/chemistry/Chemistry-Recipes.dm:4288]: are you sure result amount is 0

File: code/modules/chemistry/Reagents-Misc.dm [hunk 4]
```diff
@@ -3984,6 +3984,28 @@ datum
...
+			reaction_mob(var/mob/M, var/method=TOUCH, var/volume_passed)
+				. = ..()
+				if(!volume_passed)
+					return
+				if(!isliving(M)) // Corpses aren't well known for their hygiene
+					return
+				if(issilicon(M)) // Its not that type of oil
+					return
+				var/cleaning_amt = volume_passed
+				var/mob/living/carbon/human/H = M
```
Comment (ZeWaka) [inline code/modules/chemistry/Reagents-Misc.dm:4006]: how do you know it's a human

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Pull request closed without merging**